<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="brockad" />

<meta name="date" content="2020-09-17" />

<title>populationgenetics</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BioQ6Tutorials</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/brockad/BioQ6Tutorials">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">populationgenetics</h1>
<h4 class="author">brockad</h4>
<h4 class="date">2020-09-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>BioQ6Tutorials/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200915code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200915)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200915code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200915)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrockadBioQ6Tutorialstree555bd41804a09e1330b23356f71b7c661920f7c6targetblank555bd41a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brockad/BioQ6Tutorials/tree/555bd41804a09e1330b23356f71b7c661920f7c6" target="_blank">555bd41</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrockadBioQ6Tutorialstree555bd41804a09e1330b23356f71b7c661920f7c6targetblank555bd41a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/brockad/BioQ6Tutorials/tree/555bd41804a09e1330b23356f71b7c661920f7c6" target="_blank">555bd41</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSE116500_Limma_adj_4.csv.gz
    Ignored:    data/bdata.20130222.mhci.txt

Untracked files:
    Untracked:  analysis/populationgenetics.Rmd
    Untracked:  data/H938_Euro_chr12.geno
    Untracked:  data/H938_Euro_chr15.geno
    Untracked:  data/H938_Euro_chr2.geno
    Untracked:  data/H938_Euro_chr5.geno
    Untracked:  data/H938_Euro_chr6.geno
    Untracked:  data/H938_chr12.geno
    Untracked:  data/H938_chr15.geno
    Untracked:  data/H938_chr2.geno
    Untracked:  data/H938_chr5.geno
    Untracked:  data/H938_chr6.geno

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>-- Attaching packages ------------------------------------------------------------------------------- tidyverse 1.3.0 --</code></pre>
<pre><code>v ggplot2 3.3.2     v purrr   0.3.4
v tibble  3.0.3     v dplyr   1.0.2
v tidyr   1.1.2     v stringr 1.4.0
v readr   1.3.1     v forcats 0.5.0</code></pre>
<pre><code>-- Conflicts ---------------------------------------------------------------------------------- tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(reshape2)</code></pre>
<pre><code>
Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    smiths</code></pre>
<pre class="r"><code>data &lt;- c(&quot;H938_Euro_chr12.geno&quot;,
          &quot;H938_Euro_chr15.geno&quot;,
          &quot;H938_Euro_chr2.geno&quot;,
          &quot;H938_Euro_chr5.geno&quot;,
          &quot;H938_Euro_chr6.geno&quot;,
          &quot;H938_chr12.geno&quot;, 
          &quot;H938_chr15.geno&quot;, 
          &quot;H938_chr2.geno&quot;,
          &quot;H938_chr5.geno&quot;, 
          &quot;H938_chr6.geno&quot;) #Euro_chr2 for 

#g &lt;- read.table(paste0(&quot;data/&quot;,data[1]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[2]), header = TRUE)
g &lt;- read.table(paste0(&quot;data/&quot;,data[3]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[4]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[5]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[6]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[7]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[8]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[9]), header = TRUE)
#g &lt;- read.table(paste0(&quot;data/&quot;,data[10]), header = TRUE)

head(g)</code></pre>
<pre><code>  CHR        SNP A1 A2 nA1A1 nA1A2 nA2A2
1   2  rs4637157  G  A     2    21   101
2   2 rs13423995  G  A     0    10   114
3   2 rs13386087  A  C     0    10   114
4   2 rs11900053  T  C     2    21   101
5   2 rs11542478  G  T     3    21   100
6   2  rs6758948  C  T     0    15   109</code></pre>
<pre class="r"><code>#Calculate number of observations in each individual
g &lt;- g %&gt;%
  mutate(nObs = nA1A1 + nA1A2 + nA2A2)
#Question: Do most SNPS have complete data?
summary(g$nObs)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  108.0   124.0   124.0   123.9   124.0   124.0 </code></pre>
<pre class="r"><code>qplot(nObs, data = g)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/populationgenetics.Rmd/Package%20Loading%20and%20getting%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#Answer: Yes

#Question what is lowest count...?
#If 98.5 % of missingness of 938 expected minimum is 923
938*0.985</code></pre>
<pre><code>[1] 923.93</code></pre>
<pre class="r"><code>min(g$nObs)</code></pre>
<pre><code>[1] 108</code></pre>
<pre class="r"><code>#Answer: 887, this number is in rough agreement.

#Calculating genotype and allele frequencies

g &lt;- mutate(g, p11 = nA1A1/nObs, p12 = nA1A2/nObs, p22 = nA2A2/nObs)
g &lt;- mutate(g, p1 = p11 + 0.5*p12, p2 = p22 + 0.5*p12)

qplot(p1,p2, data = g)</code></pre>
<p><img src="figure/populationgenetics.Rmd/Package%20Loading%20and%20getting%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>model &lt;- lm(g$p1~g$p2)
summary(model)</code></pre>
<pre><code>
Call:
lm(formula = g$p1 ~ g$p2)

Residuals:
       Min         1Q     Median         3Q        Max 
-4.000e-16 -3.000e-16 -2.000e-16  0.000e+00  8.119e-12 

Coefficients:
              Estimate Std. Error    t value Pr(&gt;|t|)    
(Intercept)  1.000e+00  7.958e-16  1.257e+15   &lt;2e-16 ***
g$p2        -1.000e+00  1.007e-15 -9.935e+14   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 3.508e-14 on 53650 degrees of freedom
Multiple R-squared:      1, Adjusted R-squared:      1 
F-statistic: 9.87e+29 on 1 and 53650 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>#Equation is p2 = -p1 + 1 ; p1 = -p2 + 1

#Plotting genotype on allele frequencies

gTidy &lt;- select(g, c(p1,p11,p12,p22)) %&gt;% melt(id=&#39;p1&#39;,value.name=&quot;Genotype.Proportion&quot;)
head(gTidy)</code></pre>
<pre><code>          p1 variable Genotype.Proportion
1 0.10080645      p11          0.01612903
2 0.04032258      p11          0.00000000
3 0.04032258      p11          0.00000000
4 0.10080645      p11          0.01612903
5 0.10887097      p11          0.02419355
6 0.06048387      p11          0.00000000</code></pre>
<pre class="r"><code>ggplot(gTidy) + geom_point(aes(x = p1,
                               y = Genotype.Proportion, 
                               color = variable,
                               shape = variable))</code></pre>
<p><img src="figure/populationgenetics.Rmd/Package%20Loading%20and%20getting%20data-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>model_quad &lt;- lm(gTidy$Genotype.Proportion ~ gTidy$p1 + gTidy$variable)
summary(model_quad)</code></pre>
<pre><code>
Call:
lm(formula = gTidy$Genotype.Proportion ~ gTidy$p1 + gTidy$variable)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.44144 -0.07449 -0.00998  0.12077  0.37308 

Coefficients:
                    Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)        7.449e-02  9.651e-04   77.18   &lt;2e-16 ***
gTidy$p1          -2.494e-14  2.810e-03    0.00        1    
gTidy$variablep12  2.241e-01  1.036e-03  216.40   &lt;2e-16 ***
gTidy$variablep22  5.524e-01  1.036e-03  533.46   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1696 on 160952 degrees of freedom
Multiple R-squared:  0.6415,    Adjusted R-squared:  0.6414 
F-statistic: 9.598e+04 on 3 and 160952 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>ggplot(gTidy)+
  geom_point(aes(x=p1,y=Genotype.Proportion,color=variable,shape=variable))+
  stat_function(fun=function(p) p^2, geom=&quot;line&quot;, colour=&quot;red&quot;,size=2.5) +
  stat_function(fun=function(p) 2*p*(1-p), geom=&quot;line&quot;, colour=&quot;green&quot;,size=2.5) +
  stat_function(fun=function(p) (1-p)^2, geom=&quot;line&quot;, colour=&quot;blue&quot;,size=2.5)</code></pre>
<p><img src="figure/populationgenetics.Rmd/Package%20Loading%20and%20getting%20data-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>Testing Hardy Weinberg:</p>
<pre class="r"><code>g &lt;- mutate(g, X2 = (nA1A1-nObs*p1^2)^2 /(nObs*p1^2) + 
            (nA1A2-nObs*2*p1*p2)^2 / (nObs*2*p1*p2) + 
            (nA2A2-nObs*p2^2)^2 / (nObs*p2^2))
g &lt;- mutate(g,pval = 1-pchisq(X2,1))

head(g$pval)</code></pre>
<pre><code>[1] 0.4635295 0.6398709 0.6398709 0.4635295 0.1566482 0.4734484</code></pre>
<pre class="r"><code>dim(as.matrix(g$pval))</code></pre>
<pre><code>[1] 53652     1</code></pre>
<pre class="r"><code>sum(g$pval &lt; 0.05, na.rm = TRUE)</code></pre>
<pre><code>[1] 2753</code></pre>
<pre class="r"><code>14314/19560</code></pre>
<pre><code>[1] 0.7317996</code></pre>
<pre class="r"><code>library(qvalue)
qobj &lt;- qvalue(p = g$pval)
qplot(pval, data = g)</code></pre>
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>Warning: Removed 3843 rows containing non-finite values (stat_bin).</code></pre>
<p><img src="figure/populationgenetics.Rmd/Hardy%20Weinberg-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist(qobj)</code></pre>
<p><img src="figure/populationgenetics.Rmd/Hardy%20Weinberg-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot expected vs observed heterozygosity
qplot(2*p1*(1-p1), p12, data = g) + geom_abline(intercept = 0,
                                                slope=1,
                                                color=&quot;red&quot;,
                                                size=1.5)</code></pre>
<p><img src="figure/populationgenetics.Rmd/Hardy%20Weinberg-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mean(((2 * g$p1 * (1-g$p1)) - g$p12) / (2 * g$p1 * (1-g$p1)), na.rm = TRUE)</code></pre>
<pre><code>[1] 0.01042792</code></pre>
<pre class="r"><code>g &lt;- mutate(g, F = (2*p1*(1-p1)-p12) / (2*p1*(1-p1)))
mean(g$F)</code></pre>
<pre><code>[1] NaN</code></pre>
<pre class="r"><code>plot(g$F, xlab = &quot;SNP number&quot;)</code></pre>
<p><img src="figure/populationgenetics.Rmd/Hardy%20Weinberg-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>movingavg &lt;- function(x, n=5){stats::filter(x, rep(1/n,n), sides = 2)}
avgF &lt;- movingavg(g$F)
plot(avgF, xlab=&quot;SNP number&quot;)</code></pre>
<p><img src="figure/populationgenetics.Rmd/Hardy%20Weinberg-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>outlier=which (movingavg(g$F) == max(movingavg(g$F),na.rm=TRUE))
outlier=which.max (avgF)
g[outlier,]</code></pre>
<pre><code>      CHR       SNP A1 A2 nA1A1 nA1A2 nA2A2 nObs        p11       p12       p22
32850   2 rs1438887  A  C     3    16   105  124 0.02419355 0.1290323 0.8467742
              p1        p2       X2       pval         F
32850 0.08870968 0.9112903 5.056231 0.02453755 0.2019308</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] qvalue_2.20.0   reshape2_1.4.4  forcats_0.5.0   stringr_1.4.0  
 [5] dplyr_1.0.2     purrr_0.3.4     readr_1.3.1     tidyr_1.1.2    
 [9] tibble_3.0.3    ggplot2_3.3.2   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.0 xfun_0.17        splines_4.0.2    haven_2.3.1     
 [5] colorspace_1.4-1 vctrs_0.3.4      generics_0.0.2   htmltools_0.5.0 
 [9] yaml_2.2.1       blob_1.2.1       rlang_0.4.7      later_1.1.0.1   
[13] pillar_1.4.6     withr_2.2.0      glue_1.4.2       DBI_1.1.0       
[17] dbplyr_1.4.4     modelr_0.1.8     readxl_1.3.1     plyr_1.8.6      
[21] lifecycle_0.2.0  munsell_0.5.0    gtable_0.3.0     workflowr_1.6.2 
[25] cellranger_1.1.0 rvest_0.3.6      evaluate_0.14    labeling_0.3    
[29] knitr_1.29       httpuv_1.5.4     fansi_0.4.1      broom_0.7.0     
[33] Rcpp_1.0.5       promises_1.1.1   backports_1.1.9  scales_1.1.1    
[37] jsonlite_1.7.1   farver_2.0.3     fs_1.5.0         hms_0.5.3       
[41] digest_0.6.25    stringi_1.5.3    rprojroot_1.3-2  grid_4.0.2      
[45] cli_2.0.2        tools_4.0.2      magrittr_1.5     crayon_1.3.4    
[49] pkgconfig_2.0.3  ellipsis_0.3.1   xml2_1.3.2       reprex_0.3.0    
[53] lubridate_1.7.9  assertthat_0.2.1 rmarkdown_2.3    httr_1.4.2      
[57] rstudioapi_0.11  R6_2.4.1         git2r_0.27.1     compiler_4.0.2  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
